Author: Alessio Treglia <alessio@debian.org>
Description:
 Correct linking order to pass LDFLAGS to the linker.
 Set installation prefix to /usr.
 Install data files in /usr/share/gmtp rather than /usr/share/gMTP.
 Show compilation and linking messages.
 A bit of clean-up.
Forwarded: not-needed
---
 Makefile |   58 ++++++++++++++++++++++++++--------------------------------
 1 file changed, 26 insertions(+), 32 deletions(-)

--- gmtp.orig/Makefile
+++ gmtp/Makefile
@@ -1,15 +1,10 @@
 # gMTP Sync tool
 
-PKG_NAME = gMTP
-PREFIX = /usr/local
+PKG_NAME = gmtp
+PREFIX = /usr
 VER = 0.6
 # Note: If you update above, please update the config.h and pkginfo file as well.
 
-PKG = gMTP
-ARCH = i386
-PKGFILE = $(PKG)-$(VER)-$(ARCH).pkg
-TARFILE = $(PKG_NAME)-$(VER)-$(ARCH).tar
-
 UNAME = $(shell uname)
 
 # See what OS we are, and set things for Solaris, otherwise use a default
@@ -27,7 +22,7 @@ GCONFTOOL = gconftool-2
 TAR = tar
 
 CFLAGS += -c -O 
-LD_FLAGS += 
+LDFLAGS +=
 LIBS +=
 
 GTK_CFLAGS = `pkg-config --cflags gtk+-2.0 gconf-2.0 libmtp id3tag`
@@ -37,38 +32,37 @@ objects = main.o mtp.o interface.o callb
 headers = main.h mtp.h interface.h callbacks.h prefs.h dnd.h config.h
 
 gMTP:	$(objects)
-	@echo "B: gMTP"
-	@$(CC) -o gMTP $(LD_FLAGS) $(LIBS) $(GTK_LDFLAGS) $(objects)
+	$(CC) $(LDFLAGS) $(objects) $(GTK_LDFLAGS) $(LIBS) -o gMTP
 
 
 .c.o: $(headers)
-	@echo "C: "$<
-	@$(CC) $(CFLAGS) $(GTK_CFLAGS) -o $@  $<
+	$(CC) $(CFLAGS) $(GTK_CFLAGS) -o $@  $<
 
 
 install: gMTP
-	$(INSTALL) -d $(PREFIX)
-	$(INSTALL) -d $(PREFIX)/bin
-	$(INSTALL) -d $(PREFIX)/share
-	$(INSTALL) -d $(PREFIX)/share/$(PKG_NAME)
-	$(INSTALL) -d $(PREFIX)/share/applications
-	$(INSTALL) -d $(PREFIX)/share/pixmaps
-	$(INSTALL) -m 755 gMTP $(PREFIX)/bin
-	$(INSTALL) -m 644 icon.png $(PREFIX)/share/$(PKG_NAME)
-	$(INSTALL) -m 644 icon-16.png $(PREFIX)/share/$(PKG_NAME)
-	$(INSTALL) -m 644 stock-about-16.png $(PREFIX)/share/$(PKG_NAME)
-	$(INSTALL) -m 644 README $(PREFIX)/share/$(PKG_NAME)
-	$(INSTALL) -m 644 COPYING $(PREFIX)/share/$(PKG_NAME)
-	$(INSTALL) -m 644 ChangeLog $(PREFIX)/share/$(PKG_NAME)
-	$(INSTALL) -m 644 AUTHORS $(PREFIX)/share/$(PKG_NAME)
-	$(INSTALL) -m 644 gMTP.desktop $(PREFIX)/share/applications
-	$(INSTALL) -m 644 icon.png $(PREFIX)/share/pixmaps
-	mv $(PREFIX)/share/pixmaps/icon.png $(PREFIX)/share/pixmaps/gMTPicon.png
-	GCONF_CONFIG_SOURCE=`$(GCONFTOOL) --get-default-source` $(GCONFTOOL) --makefile-install-rule gMTP.schema
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/$(PKG_NAME)
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/applications
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/pixmaps
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/doc
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/doc/$(PKG_NAME)
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/gconf
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/share/gconf/schemas
+	$(INSTALL) -m 755 gMTP $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) -m 644 icon.png $(DESTDIR)$(PREFIX)/share/$(PKG_NAME)
+	$(INSTALL) -m 644 icon-16.png $(DESTDIR)$(PREFIX)/share/$(PKG_NAME)
+	$(INSTALL) -m 644 stock-about-16.png $(DESTDIR)$(PREFIX)/share/$(PKG_NAME)
+	$(INSTALL) -m 644 README $(DESTDIR)$(PREFIX)/share/doc/$(PKG_NAME)
+	$(INSTALL) -m 644 gMTP.desktop $(DESTDIR)$(PREFIX)/share/applications
+	$(INSTALL) -m 644 icon.png $(DESTDIR)$(PREFIX)/share/pixmaps
+	$(INSTALL) -m 644 gMTP.schema $(DESTDIR)$(PREFIX)/share/gconf/schemas
+	mv $(DESTDIR)$(PREFIX)/share/pixmaps/icon.png $(DESTDIR)$(PREFIX)/share/pixmaps/gMTPicon.png
+	mv $(DESTDIR)$(PREFIX)/share/gconf/schemas/gMTP.schema $(DESTDIR)$(PREFIX)/share/gconf/schemas/gmtp.schemas
 
 clean:
-	@echo "CLEAN: core gMTP *.o *~ "$(objects)
-	@-rm -f $(objects) core gMTP *.o *~ $(PKGFILE).gz
+	rm -f $(objects) core gMTP *.o *~ $(PKGFILE).gz
 
 dist:
 	-rm -f $(objects) core gMTP *.o *~
